File: programmers/js/level01/77484.js
// 로또의 최고 순위와 최저 순위
function solution(lottos, win_nums) {
  let count0 = 0;
  let countWinNums = 0;
  
  lottos.forEach(num => num ? 
    win_nums.includes(num) && countWinNums++ : count0++
  );
  
  return countWinNums ? 
    [7 - countWinNums - count0, 7 - countWinNums] : 
    count0 ? [1, 6] : [6, 6];
}

// 다른 풀이
// function solution(lottos, win_nums) {
//   const rank = [6, 6, 5, 4, 3, 2, 1];

//   let minCount = lottos.filter(v => win_nums.includes(v)).length;
//   let zeroCount = lottos.filter(v => !v).length;

//   const maxCount = minCount + zeroCount;

//   return [rank[maxCount], rank[minCount]];
// }

File: programmers/java/level01/77484.java
// 로또의 최고 순위와 최저 순위

class Solution {
    boolean includesInt(int[] array, int value) {
        for (int num : array) {
            if (num == value) return true;
        }    
        
        return false;
    }
    
    int getRank(int number) {
        switch (number) {
            case 6: 
                return 1;
            case 5: 
                return 2;
            case 4:
                return 3;
            case 3:
                return 4;
            case 2:
                return 5;
            default:
                return 6;
        }
    }
    
    public int[] solution(int[] lottos, int[] win_nums) {
        int countMatched = 0;
        int countZero = 0;
        
        for (int num : lottos) {
            if (num == 0) countZero++;
            else if (includesInt(win_nums, num)) countMatched++;
        }
        
        return new int[]{getRank(countMatched + countZero), getRank(countMatched)};
    }
}